kubectl-minio/cmd/cmd-operator-create.go

```diff
@@ -0,0 +1,200 @@
+	serviceAccount string
+}
+
+func newOperatorCreateCmd(out io.Writer, errOut io.Writer) *cobra.Command {
+	i := &operatorCreateCmd{out: out, errOut: errOut}
+
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create MinIO Operator deployment",
+		Long:  operatorCreateDesc,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 0 {
+				return errors.New("this command does not accept arguments")
+			}
+			return i.run()
+		},
+	}
+
+	f := cmd.Flags()
+	f.StringVarP(&i.image, "image", "i", helpers.DefaultOperatorImage, "MinIO Operator image")
+	f.StringVarP(&i.ns, "namespace", "n", helpers.DefaultNamespace, "If present, the namespace scope for this request ")
+	f.StringVarP(&i.serviceAccount, "service-account", "", helpers.DefaultServiceAccount, "ServiceAccount for MinIO Operator")
+	f.StringVarP(&i.clusterDomain, "cluster-domain", "d", helpers.DefaultClusterDomain, "Cluster domain of the Kubernetes cluster")
+	f.StringVarP(&i.nsToWatch, "namespace-to-watch", "", "", "Namespace where MinIO Operator looks for MinIO Instances, leave empty for all namespaces")
+
+	return cmd
+}
+
+// run initializes local config and installs MinIO Operator to Kubernetes cluster.
+func (i *operatorCreateCmd) run() error {
+	sch := runtime.NewScheme()
+	_ = scheme.AddToScheme(sch)
+	_ = apiextensionv1.AddToScheme(sch)
+	decode := serializer.NewCodecFactory(sch).UniversalDeserializer().Decode
+
+	crd, err := pkger.Open("/static/crd.yaml")
+	if err != nil {
+		return err
+	}
+	defer crd.Close()
+	obj, _, err := decode(helpers.StreamToByte(crd), nil, nil)
+	if err != nil {
+		return err
+	}
+
+	crdObj := obj.(*apiextensionv1.CustomResourceDefinition)
+
+	cr, err := pkger.Open("/static/cluster-role.yaml")
+	if err != nil {
+		return err
+	}
+	defer cr.Close()
+	obj, _, err = decode(helpers.StreamToByte(cr), nil, nil)
+	if err != nil {
+		return err
+	}
+
+	crObj := obj.(*rbacv1.ClusterRole)
+	sa := resources.NewServiceAccountForOperator(i.serviceAccount, i.ns)
+	crb := resources.NewCluterRoleBindingForOperator(i.ns, i.serviceAccount)
+	d := resources.NewDeploymentForOperator(i.ns, i.serviceAccount, i.image, i.clusterDomain, i.nsToWatch)
+
+	client, err := helpers.GetKubeClient()
+	if err != nil {
+		return err
+	}
+	extclient, err := helpers.GetKubeExtensionClient()
+	if err != nil {
+		return err
+	}
+	if err = createCRD(extclient, crdObj); err != nil {
+		return err
+	}
+	if err = createCR(client, crObj); err != nil {
+		return err
+	}
+	if err = createSA(client, sa); err != nil {
+		return err
+	}
+	if err = createClusterRB(client, crb); err != nil {
+		return err
+	}
+	if err = createDeployment(client, d); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createCRD(client *apiextension.Clientset, crd *apiextensionv1.CustomResourceDefinition) error {
+	_, err := client.ApiextensionsV1().CustomResourceDefinitions().Create(context.TODO(), crd, v1.CreateOptions{})
+	if err != nil {
+		if !kerrors.IsAlreadyExists(err) {
```

Comment:
in this case print

```
fmt.Printf("CustomResourceDefinition %s: unconfigured\n", crd.ObjectMeta.Name)
```

Reply:
This case, the CRD is already present so we just move forward to create Operator deployment. IMO there is no point to print something in this case.

Reply:
We print that we are not creating anything i.e unconfigured currently it prints creating again because it ignores an existing operator

Reply:
done.